test/functional/system/rounds.js

```diff
@@ -0,0 +1,1584 @@
+					if (tick.isLastBlockOfRound) {
+						return expect(tick.before.delegatesForList).to.not.deep.equal(
+							tick.after.delegatesForList
+						);
+					}
+
+					return expect(tick.before.delegatesForList).to.deep.equal(
+						tick.after.delegatesForList
+					);
+				});
+
+				it('delegates list should be the same for same round', () => {
+					if (
+						(tick.isLastBlockOfRound &&
+							!_.isEqual(
+								tick.before.delegatesForList,
+								tick.after.delegatesForList
+							)) ||
+						tick.isRoundChanged
+					) {
+						return expect(tick.before.delegatesList).to.not.deep.equal(
+							tick.after.delegatesList
+						);
+					} else if (
+						tick.isLastBlockOfRound &&
+						_.isEqual(tick.before.delegatesForList, tick.after.delegatesForList)
+					) {
+						return expect(tick.before.delegatesList).to.deep.equal(
+							tick.after.delegatesList
+						);
+					}
+
+					return expect(tick.before.delegatesList).to.deep.equal(
+						tick.after.delegatesList
+					);
+				});
+
+				it('accounts table states should match expected states', done => {
+					let expected;
+
+					expected = expectedMemState(transactions, tick.before.accounts);
+
+					// Last block of round - apply round expectactions
+					if (tick.isLastBlockOfRound) {
+						expected = applyRoundRewards(expected, tick.roundBlocks);
+						expected = recalculateVoteWeights(expected, tick.voters);
+						expected = applyOutsiders(
+							expected,
+							tick.before.delegatesList,
+							tick.roundBlocks
+						);
+					}
+
+					// FIXME: Remove that nasty hack after https://github.com/LiskHQ/lisk/issues/716 is closed
+					try {
+						expect(tick.after.accounts).to.deep.equal(expected);
+					} catch (err) {
+						// When comparison of mem_accounts states fail
+						_.reduce(
+							tick.after.accounts,
+							(result, value, key) => {
+								// Clone actual and expected accounts states
+								const actualAccount = Object.assign({}, value);
+								const expectedAccount = Object.assign({}, expected[key]);
+								// Compare actual and expected states
+								if (!_.isEqual(actualAccount, expectedAccount)) {
+									// When comparison fails - calculate absolute difference of 'vote' values
+									const absoluteDiff = Math.abs(
+										new Bignum(actualAccount.vote)
+											.minus(new Bignum(expectedAccount.vote))
+											.toNumber()
+									);
+									// If absolute value is 1 beddows - pass the test, as reason is related to issue #716
+									if (absoluteDiff === 1) {
+										__testContext.debug(
+											`ERROR: Value of 'vote' for account ${key} doesn't match expectations, actual: ${
+												actualAccount.vote
+											}, expected: ${
+												expectedAccount.vote
+											}, diff: ${absoluteDiff} beddows`
+										);
+									} else {
+										// In every other case - fail the test
+										throw err;
+									}
+								}
+							},
+							[]
+						);
+					}
```

Comment:
What if we have simple logic here: 

```
tick.after.accounts = tick.after.accounts.map((account) {
   if ( Math.abs( new Bignum(account.vote).minus(expected(address).vote)).toNumber() === 1 ) {
     account.vote = new Bignum(account.vote).plus('1').toString();
   }
   return account; 
));

expect(tick.after.accounts).to.deep.equal(expected);
```

Reply:
It will not work, as `tick.after.accounts` is an object